File: src/test/java/org/acme/elasticsearch/FruitServiceTest.java
package org.acme.elasticsearch;

import io.quarkus.test.junit.QuarkusTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import javax.inject.Inject;
import java.io.IOException;
import java.util.List;

@QuarkusTest
public class FruitServiceTest {

    @Inject
    FruitService fruitService;

    @Test
    void testSearch() throws IOException {
        Fruit fruit = Fruit.builder().color("red").name("strawberry").build();
        fruitService.index(fruit);
        List<Fruit> searchResult = fruitService.searchByColor("red");
        Assertions.assertFalse(searchResult.isEmpty());
        Fruit fruitFounded = searchResult.stream().findFirst().orElse(Fruit.builder().build());
        Assertions.assertEquals(fruitFounded.color, "red");
        Assertions.assertEquals(fruitFounded.name, "strawberry");
    }
}

File: src/test/java/org/acme/elasticsearch/FruitServiceTest.kt
package org.acme.elasticsearch

import io.quarkus.test.junit.QuarkusTest
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.io.IOException
import java.util.*
import javax.enterprise.inject.Default
import javax.inject.Inject

@QuarkusTest
internal class FruitServiceTest(){

    @Inject
    @field: Default
    lateinit var fruitService: FruitService

    @Test
    @Throws(IOException::class)
    fun testSearch() {
        val id = UUID.randomUUID().toString()
        val fruit = Fruit(id, "strawberry", "red")
        fruitService.index(fruit)
        val searchResult = fruitService.searchByColor("red")
        assertFalse(searchResult.isEmpty())
        val fruitFounded = searchResult.stream().findFirst().orElse(Fruit())
        assertEquals(fruitFounded.color, "red")
        assertEquals(fruitFounded.name, "strawberry")
    }
}